runtime/jcl/common/java_lang_invoke_MethodHandleNatives.cpp

```diff
@@ -133,6 +133,12 @@ initImpl(J9VMThread *currentThread, j9object_t membernameObject, j9object_t refO
 		flags = fieldID->field->modifiers & CFR_FIELD_ACCESS_MASK;
 		flags |= MN_IS_FIELD;
 		flags |= (J9_ARE_ANY_BITS_SET(flags, J9AccStatic) ? MH_REF_GETSTATIC : MH_REF_GETFIELD) << MN_REFERENCE_KIND_SHIFT;
+		if (J9_ARE_ALL_BITS_SET(flags, J9AccFinal)) {
+			J9ROMClass *romClass = fieldID->declaringClass->romClass;
+			if (J9_ARE_ALL_BITS_SET(flags, J9AccStatic) || J9ROMCLASS_IS_HIDDEN(romClass) || J9ROMCLASS_IS_RECORD(romClass)) {
```

Comment:
```suggestion
			if (J9_ARE_ALL_BITS_SET(flags, J9AccStatic) && (J9ROMCLASS_IS_HIDDEN(romClass) || J9ROMCLASS_IS_RECORD(romClass))) {
```

Should that be isStatic and a hidden Class or a record?  We don't want all static fields on all classes to be trusted....

Does this apply to other finals in say, the java.lang.invoke package as well?

Reply:
this matchesthe behavior of RI's impl.
This check appplies to all classes, ie all static finals are considered trusted, also any final fields in a hidden/record class is considered trusted.

This flag is only used by MH unreflect code to check if write access is allowed to the field (I don't think this affect the VM's understanding of a field is trusted or not)

Reply:
I did some digging to double check this and you're right.  There was a OpenJDK CSR that clarified this has actually long been their behaviour.